setup.py

```diff
@@ -45,7 +48,7 @@ def main():
     install_requires=[
         'dill>=0.3.2',
     ],
-    python_requires='>=3.6, <3.9',
+    python_requires='>=3.6, <=3.11',
```

Comment:
3.11 will be released soon, but it is still in preview. I think it would be better to avoid explicit support of 3.11 for now.
```suggestion
    python_requires='>=3.6, <3.11',
```